### PR TITLE
DT-2295 topics integration tests

### PR DESCRIPTION
**Pull request by mikehalmamoj**

### CODE REVIEW

File: test-app/helm_deploy/hmpps-template-kotlin/values.yaml [hunk 1]
```diff
@@ -30,22 +30,30 @@ generic-service:
...
+      HMPPS_SQS_QUEUES_OUTBOUNDQUEUE_DLQ_SECRET_ACCESS_KEY: "dlq_another_secret_access_key"
+      HMPPS_SQS_QUEUES_OUTBOUNDQUEUE_DLQ_NAME: "anotherDlq"
+    inbound-topic:
+      HMPPS_SQS_TOPICS_INBOUNDTOPIC_ACCESS_KEY_ID: "access_key_id"
+      HMPPS_SQS_TOPICS_INBOUNDTOPIC_SECRET_ACCESS_KEY: "secret_access_key"
+      HMPPS_SQS_TOPICS_INBOUNDTOPIC_TOPIC_ARN: "topic_arn"
+    outbound-topic:
+      HMPPS_SQS_TOPICS_OUTBOUNDTOPIC_ACCESS_KEY_ID: "hmpps_access_key_id"
+      HMPPS_SQS_TOPICS_OUTBOUNDTOPIC_SECRET_ACCESS_KEY: "hmpps_secret_access_key"
+      HMPPS_SQS_TOPICS_OUTBOUNDTOPIC_TOPIC_ARN: "hmpps_topic_arn"
```
Comment (mikehalmamoj) [inline test-app/helm_deploy/hmpps-template-kotlin/values.yaml:56]: change values to reflect outbound / inbound secrets and dlqs etc.

File: test-app/src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/service/MessageListener.kt [hunk 2]
```diff
@@ -1,22 +1,36 @@
...
 @Service
-class MessageListener(private val messageService: MessageService) {
+class InboundMessageListener(private val inboundMessageService: InboundMessageService, private val objectMapper: ObjectMapper) {
 
-  @JmsListener(destination = "mainqueue", containerFactory = "hmppsQueueContainerFactoryProxy")
-  fun processMessage(message: String?) {
-    messageService.handleMessage(message ?: "empty message received for mainqueue")
+  @JmsListener(destination = "inboundqueue", containerFactory = "hmppsQueueContainerFactoryProxy")
+  fun processMessage(rawMessage: String?) {
+    val (Message) = objectMapper.readValue(rawMessage, Message::class.java)
```
Comment (mikehalmamoj) [inline test-app/src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/service/MessageListener.kt:12]: rename to `message`

File: test-app/src/test/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/integration/HmppsEventProcessingTest.kt [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    val event = HmppsEvent("event-id", "OFFENDER_MOVEMENT-RECEPTION", "some event contents")
+    inboundSnsClient.publish(
+      PublishRequest(inboundTopicArn, gsonString(event))
+        .withMessageAttributes(
+          mapOf("eventType" to MessageAttributeValue().withDataType("String").withStringValue(event.type))
+        )
+    )
+
+    await untilCallTo { outboundTestSqsClient.countMessagesOnQueue(outboundTestQueueUrl) } matches { it == 1 }
+    val (Message) = objectMapper.readValue(outboundTestSqsClient.receiveMessage(outboundTestQueueUrl).messages[0].body, Message::class.java)
```
Comment (mikehalmamoj) [inline test-app/src/test/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/integration/HmppsEventProcessingTest.kt:27]: rename to message

**Review by karenmillermoj — APPROVED** [review]
lgtm

**Pull request merged**